lib/std/zon/stringify.zig

```diff
@@ -0,0 +1,2088 @@
+        /// Serialize a value, similar to `serializeMaxDepth`.
+        pub fn valueMaxDepth(
+            self: *Self,
+            val: anytype,
+            options: ValueOptions,
+            depth: usize,
+        ) (Writer.Error || error{ExceededMaxDepth})!void {
+            try checkValueDepth(val, depth);
+            return self.valueArbitraryDepth(val, options);
+        }
+
+        /// Serialize a value, similar to `serializeArbitraryDepth`.
+        pub fn valueArbitraryDepth(
+            self: *Self,
+            val: anytype,
+            options: ValueOptions,
+        ) Writer.Error!void {
+            switch (@typeInfo(@TypeOf(val))) {
+                .int => |int_info| if (options.emit_utf8_codepoints and
+                    int_info.signedness == .unsigned and
+                    int_info.bits <= 21 and std.unicode.utf8ValidCodepoint(val))
+                {
+                    self.utf8Codepoint(val) catch |err| switch (err) {
+                        error.InvalidCodepoint => unreachable, // Already validated
+                        else => |e| return e,
+                    };
+                } else {
+                    try self.int(val);
+                },
+                .comptime_int => if (options.emit_utf8_codepoints and
+                    val > 0 and
+                    val <= std.math.maxInt(u21) and
+                    std.unicode.utf8ValidCodepoint(val))
+                {
+                    self.utf8Codepoint(val) catch |err| switch (err) {
+                        error.InvalidCodepoint => unreachable, // Already validated
+                        else => |e| return e,
+                    };
+                } else {
+                    try self.int(val);
+                },
+                .float, .comptime_float => try self.float(val),
+                .bool, .null => try std.fmt.format(self.writer, "{}", .{val}),
+                .enum_literal => {
+                    try self.writer.writeByte('.');
+                    try self.ident(@tagName(val));
+                },
+                .@"enum" => |@"enum"| if (@"enum".is_exhaustive) {
+                    try self.writer.writeByte('.');
+                    try self.ident(@tagName(val));
+                } else {
+                    @compileError(
+                        @typeName(@TypeOf(val)) ++ ": cannot stringify non-exhaustive enums",
+                    );
+                },
+                .void => try self.writer.writeAll("{}"),
+                .pointer => |pointer| {
+                    const child_type = switch (@typeInfo(pointer.child)) {
+                        .array => |array| array.child,
+                        else => if (pointer.size != .slice) @compileError(
+                            @typeName(@TypeOf(val)) ++ ": cannot stringify pointer to this type",
+                        ) else pointer.child,
+                    };
+                    if (child_type == u8 and !options.emit_strings_as_containers) {
```

Comment:
The string rules should be consistent between parsing and stringification; so, since `std.zon.parse` checks the type doesn't have a non-zero sentinel, so should `std.zon.stringify`.